src/main/java/org/mitre/synthea/engine/Generator.java

```diff
@@ -334,52 +351,250 @@ private synchronized void writeToConsole(Person person, int index, long time, bo
     }
   }
 
-  private long setDemographics(Person person, Demographics city) {
-    person.attributes.put(Person.CITY, city.city);
-    person.attributes.put(Person.STATE, city.state);
-    
-    String race = city.pickRace(person.random);
-    person.attributes.put(Person.RACE, race);
-    String ethnicity = city.ethnicityFromRace((String)person.attributes.get(Person.RACE), person);
-    person.attributes.put(Person.ETHNICITY, ethnicity);
-    String language = city.languageFromEthnicity((String) person.attributes.get(Person.ETHNICITY),
-        person);
-    person.attributes.put(Person.FIRST_LANGUAGE, language);
-
-    String gender = city.pickGender(person.random);
-    if (gender.equalsIgnoreCase("male") || gender.equalsIgnoreCase("M")) {
-      gender = "M";
+  private long setDemographics(Person person, Demographics city) throws IOException {
+    if (Demographics.householdsMode() == "true") {
```

Comment:
Compare strings using .equals() not == . This comparison might work out sometimes but it's generally not going to work how you expect. Also consider creating a new boolean field for this; take a look at the "onlyDeadPatients" field for an example.